### PR TITLE
Only call getTerminalMap() when terminal is actually requested

### PR DESCRIPTION
**Pull request by maihde**
When fillFromStat() is called, it uses getTerminalMap() to
resolve the terminal, but this can use a significant amount
of cpu utilization simply to be thrown away with most of
the calls.  By moving getTerminaMap() into the Terminal()
method the performance improves significantly (50%+).

### CODE REVIEW

File: vendor/github.com/shirou/gopsutil/process/process_linux.go [hunk 1]
```diff
@@ -176,10 +176,12 @@ func (p *Process) Gids() ([]int32, error) {
 
 // Terminal returns a terminal which is associated with the process.
 func (p *Process) Terminal() (string, error) {
-	terminal, _, _, _, _, err := p.fillFromStat()
+	t, _, _, _, _, err := p.fillFromStat()
+	termmap, err := getTerminalMap()
```
Comment (schmichael) [inline vendor/github.com/shirou/gopsutil/process/process_linux.go]: Check the error return from fillFromStat before calling getTerminalMap

**Comment by schmichael** [comment]
This PR changes a vendored dependency so it will need to be submitted to the upstream project instead: https://github.com/shirou/gopsutil

**Comment by schmichael** [comment]
Thanks for the PR though and please feel free to post a link to the upstream PR in here! We'd love to get this improvement landed!

File: vendor/github.com/shirou/gopsutil/process/process_linux.go (hunk 1 above)
Comment (maihde) [reply vendor/github.com/shirou/gopsutil/process/process_linux.go]: This proposed change has been included in the upstream vendor PR.

**Pull request closed without merging**